src/hotspot/share/gc/parallel/parallelScavengeHeap.cpp

```diff
@@ -428,18 +422,15 @@ HeapWord* ParallelScavengeHeap::satisfy_failed_allocation(size_t size, bool is_t
     HeapMaximumCompactionInterval = old_interval;
   }
 
-  result = expand_heap_and_allocate(size, is_tlab);
-  if (result != nullptr) {
-    return result;
+  if (check_gc_overhead_limit()) {
```

Comment:
What is the effect of calling this method twice? Line 400 above, and then again here on line 425. Does that increment `_gc_overhead_counter` twice? More reason why i think the name is confusing.